### DISCUSSION TITLE
Which dev board is for the pinecilV1?

### DISCUSSION THREAD

**Discussion by Spagett1** [Ralim/IronOS#1571] [Q&A]
Gday, im writing a flashing tool for the pinecil and i have been told by river that you used one of these https://www.seeedstudio.com/Sipeed-Longan-Nano-RISC-V-GD32VF103CBT6-Development-Board-p-4205.html but i was wondering which version of that board to get as they have both a V1 and V1.1 or if either would work?

Thanks.

**Comment by River-Mochi** [comment]
@Ralim  any Idea if this Aliexpress one will work for V1 firmware updater testing?

**https://www.aliexpress.us/item/3256803280749848.html**

The seed studio is out of stock and they prob charge shipping too.

**Comment by robertlipe** [comment]
I can't speak to what he used, but I understand that software ecosystem reasonably well.

Sipeed Longan is the same SoC that's used in PinecilV1. If you're just interested in debugging/developing similar code,  a board like that or its near-twin (the VDCs are slightly different) the LilyGo T-Display https://www.aliexpress.us/item/2255800412041558.html (they have a bunch of boards named T-display - get the one with GD32V) are handy because they have a screen, have zillions of examples on Github, are cheap to burn up (oopsy!) and have good developer support from Nuclei, the maker of the core. The pinouts make it easy to attach JTAG or serial connections.

If you want to write code on Pinecil itself, look at https://pine64.com/product/pinecil-break-out-board/

A suitably motivated developer could probably run a substantial part of the IronOS on a Longan or T-Display by reconfiguring the button code, doing away with all that actual "soldering iron" stuff. The interface to USB could be kept intact, I think, and that's the part you care about. It's probably easier to just disable the display than to port the OLED code to the ST7735/7789.

You'd have to be pretty motivated to do this over finding a used Pinecil v1 to buy or borrow... $35 doesn't buy a lot of embedded developer time. :-)

**Comment by Ralim** [comment]
I used what I assume was the V1 as it was the only version at the time.

Basically any will work for what you want, you just need to be able to short boot to enter into bootloader + a working USB port.
For just testing you can look at a working flash being one if it disconnects from the usb bus and doesnt enumerate (i.e the code started running; and it didnt fall back into bootloader). If you corrupt the flash it should drop back into the bootloader.

Also you can validate by doing a write + read test and comparing; should get same out as went in.
The Sipeed Longan  is what I started on back when they were < $5 shipped.
